code/game/objects/items/weapons/implants/vorestuff_vr.dm

```diff
@@ -0,0 +1,131 @@
+/obj/item/weapon/implant/reagent_generator
+	name = "reagent generator implant"
+	desc = "This is an implant that has attached storage and generates a reagent."
+	implant_color = "r"
+	var/obj/item/weapon/reagent_containers/glass/beaker/large/internal_storage
+	var/datum/reagent/generated_reagent = null
+	var/gen_amount = 2 //amount of reagent generated per process tick
+	var/gen_cost = 0.5 //amount of nutrient taken from the host per process tick
+	var/transfer_amount = 30 //amount transferred when using verb
+	var/usable_volume = 120
+
+	var/empty_message = "You feel as though your internal reagent implant is almost empty."
+	var/full_message = "You feel as though your internal reagent implant is full."
+	var/emote_descriptor = "tranfers something" //In format of [x] [emote_descriptor] into [container]
+	var/self_emote_descriptor = "transfer" //In format of You [self_emote_descriptor] some [generated_reagent] into [container]
+	var/random_emote = null //An emote the person with the implant may be forced to perform after a prob check, such as [X] meows.
+	var/assigned_proc = /mob/living/carbon/human/proc/use_reagent_implant
+
+/obj/item/weapon/implant/reagent_generator/tempest
+	generated_reagent = "milk"
+	usable_volume = 1000
+
+	empty_message = "Your breasts are almost completely drained!"
+	full_message = "Your teats feel heavy and swollen!"
+	emote_descriptor = "squeezes milk"
+	self_emote_descriptor = "squeeze"
+	random_emote = "moos"
+	assigned_proc = /mob/living/carbon/human/proc/use_reagent_implant_tempest
+
+/obj/item/weapon/implant/reagent_generator/New()
+	..()
+	internal_storage = new /obj/item/weapon/reagent_containers/glass/beaker/large (src)
+	internal_storage.volume = usable_volume
+	internal_storage.reagents.maximum_volume = usable_volume
+
+/obj/item/weapon/implanter/reagent_generator/New()
+	..()
+	imp = new /obj/item/weapon/implant/reagent_generator(src)
+	update()
+	return
+
+/obj/item/weapon/implanter/reagent_generator/tempest/New()
+	..()
+	imp = new /obj/item/weapon/implant/reagent_generator/tempest(src)
+	update()
+	return
+
+/obj/item/weapon/implant/reagent_generator/implanted(mob/living/carbon/source)
+	processing_objects += src
+	to_chat(source, "<span class='notice'>You implant [source] with \the [src].</span>")
+	source.verbs |= assigned_proc
+	return 1
+
+/obj/item/weapon/implant/reagent_generator/process()
+	var/before_gen
+	if(imp_in && generated_reagent)
+		before_gen = internal_storage.reagents.total_volume
+		if(internal_storage && internal_storage.reagents.total_volume < internal_storage.volume)
+			if(imp_in.nutrition >= gen_cost)
```

Comment:
Don't check for internal_storage if there's no way to remove it. Just let it runtime anyway, it's not like it'll function correctly if someone somehow removes it. Roll the if() below up into the one above.